pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -407,6 +407,16 @@ fi
 
 tmpDir=$(mktemp -t -d nixos-rebuild.XXXXXX)
 
+# macOS sets the TMPDIR to /var/folders/ab/abcdef.../A as a per-user setting
+# to improve security. However, this does not play well with SSH control paths:
+# unix_listener: path "/var/folders/ab/abcdef.../T/nixos-rebuild...." too long
+# for Unix domain socket
+# Unset TMPDIR in this case
+
+if [[ ${#tmpDir} -ge 60 ]]; then
+    tmpDir=$(TMPDIR='' mktemp -t -d nixos-rebuild.XXXXXX)
```

Comment:
This destroys the old `tmpDir` value without cleanup.
```suggestion
    rmdir $tmpDir
    tmpDir=$(TMPDIR='' mktemp -t -d nixos-rebuild.XXXXXX)
```